src/frontend/src/common/components/library/data_subview/components/CreateNSTreeModal/components/SampleFiltering/index.tsx

```diff
@@ -31,13 +156,114 @@ const SAMPLE_FILTERING_TOOLTIP_TEXT = (
+ * use of only one kind of lineage: Pango lineages. Using the generic term is
+ * mostly for ease, but we might need to approach breaking out by type if we
+ * start to support others eventually (like WHO greek letters, etc).
+ *
  * TODO Vincent -- Mar 9, 2022
  * Below `StyledInputDropdown` components are /very/ much placeholders.
  * Looked over czifui for a bit, and I couldn't find an easy way to match
  * our designs to what's current available. Those are just there to get
  * a version of this out the door, nothing longterm at all there.
  */
-export function SampleFiltering() {
+export function SampleFiltering({
+  availableLineages,
+  selectedLineages,
+  setSelectedLineages,
+}: Props) {
+  const lineageDropdownOptions = generateLineageDropdownOptions(
+    selectedLineages,
+    availableLineages
+  );
+  const lineageDropdownLabel = getLineageDropdownLabel(selectedLineages);
+  const lineageDropdownValue = getLineageDropdownValue(selectedLineages);
+
+  /**
+   * Handles setting selected lineages from user's lineage Dropdown choices.
+   *
+   * Depending on if user had started with the "All" choice selected -- that
+   * is, if no lineages had been chosen to filter to; the "All" choice can
+   * both be explicitly selected by the user or implicitly selected b/c no
+   * actual lineages are chosen -- emitted result changes. If "All" is selected
+   * when Dropdown is opened, we ignore that option choice in preference of the
+   * newly selected lineages. On the other hand, if "All" is not selected when
+   * Dropdown is opened (b/c lineages are chosen), we ignore any other lineages
+   * that get chosen if "All" is also chosen, instead preferring to reset the
+   * lineage filter back to allowing all lineages.
+   *
+   * HACK (Vince):
+   * I don't particularly like my implementation, but it's the best I could
+   * come up with over a few hours of work and thinking about it. Because "All"
+   * is not really a choice, but rather a pseudo-choice that means no lineages
+   * are selected or that the lineage filter should be reset, we get into some
+   * weird places. The Dropdown component expects every option to just be an
+   * object that is either selected or not selected. But b/c of the above, the
+   * options now have side-effects: choosing a lineage removes "All" from being
+   * selected, or choosing "All" can reset and de-select all the other options.
+   * So in addition to needing to handle this side-effect logic, we also have
+   * to avoid infinite render loops due to the side-effect setting a new
+   * `selectedLineages` upstream, which then goes down into the Dropdown, which
+   * then kicks off the onChange (b/c its controlled), which can then trigger
+   * another new side-effect handling and spiral into an infinite loop...
+   *
+   * If this winds up being refactored into something better, that would be
+   * great, but make sure to test your work pretty aggressively if you do that
```

Comment:
can we do some basic automated testing on this to make it less scary?

Reply:
I certainly like that idea, but I've been thinking about it for awhile and I can't come up with a way to write a good test for it quickly. I'm not great at testing, so maybe you've got a good way in mind or it's easier than I'm imagining? I'll Slack you about it, but I'm pretty much afraid that writing a test that would meaningfully improve its safety -- beyond the 10 minutes of manual QA I did when wrapping up that work -- would take multiple days.

Reply:
Thinking a bit more, another very different way to approach this would be to have a `preprocessSelection` prop on the Dropdown component in czifui. Instead of directly passing the selections made in the Dropdown to its `onChange` prop func, an optional `preprocessSelection` could first run on the selection -- not sure if it would be after every selection click or if it would be only on blur, both approaches seem reasonable -- and then the result of that would get handed to `onChange`.

A lot of the difficulty of this implementation came from needing to handle side-effects for how certain dropdown choices would impact other choices. A `preprocessSelection` prop func would at least bring that side-effect occurrence into the Dropdown and might make the infinite loop aspect impossible. That said, I think this is a real, real edge case use of how Dropdowns work, so not sure if it's worth that effort and muddying the Dropdown API in czifui.

Reply:
No problem, let's tackle it after we have more science test infra in 
https://app.shortcut.com/genepi/story/193963/write-a-test-to-make-sure-the-lineage-filter-dropdown-has-sane-and-functional-logic